src/charm.py

```diff
@@ -279,13 +279,27 @@ def _update_layer(self) -> None:
             try:
                 self.logger.info("Pebble plan updated with new configuration, replaning")
                 self.container.replan()
-            except ChangeError as err:
-                raise ErrorWithStatus(f"Failed to replan with error: {str(err)}", BlockedStatus)
+            except ChangeError as e:
+                raise GenericCharmRuntimeError("Failed to replan") from e
+
+    def _check_container_connection(self, container: Container) -> None:
+        """Check if connection can be made with container.
+
+        Args:
+            container: the named container in a unit to check.
+
+        Raises:
+            ErrorWithStatus if the connection cannot be made.
+        """
+        if not container.can_connect():
```

Comment:
then pls replace all container.can_connect in the code

Reply:
I think the error with container only is clear. Otherwise you can customize the function to take a message if fail. eg `f"Container is not ready. {add_msg}`. By default only not ready. If specified may be `f"Container is not ready. Failed to upload.`

Reply:
Will update, thanks!